azkaban-common/src/main/java/azkaban/project/DirectoryYamlFlowLoader.java

```diff
@@ -112,14 +113,22 @@ private void convertYamlFiles(final File projectDir) {
               + ". Duplicate nodes found or dependency undefined.");
         } else {
           final AzkabanFlow azkabanFlow = (AzkabanFlow) loader.toAzkabanNode(nodeBean);
-          final Flow flow = convertAzkabanFlowToFlow(azkabanFlow, azkabanFlow.getName(), file);
-          this.flowMap.put(flow.getId(), flow);
+          if (this.flowMap.containsKey(azkabanFlow.getName())) {
+            this.errors.add("Duplicate flows found in the project with name " + azkabanFlow
```

Comment:
Where do you check duplicate project file?

Reply:
Inside azkaban.project.FlowLoaderFactory#checkForValidProjectYamlFile. Need to check .project file first when deciding whether to create directoryFlowLoader (Flow 1.0) or directoryYamlFlowLoader (Flow 2.0). The check happens before loading project flows.